sonic_platform_base/component_base.py

```diff
@@ -103,3 +103,28 @@ def update_firmware(self, image_path):
             RuntimeError: update failed
         """
         raise NotImplementedError
+
+    def auto_update_firmware(self, image_path, boot_action):
+        """
+        Updates firmware of the component
+
+        This API performs firmware update automatically based on boot_action: it assumes firmware installation
+        and/or creating a loading task during a boot action, if needed, in a single call.
+        In case platform component requires some extra steps (apart from calling Low Level Utility)
+        to load the installed firmware (e.g, reboot, power cycle, etc.) - this will be done automatically during the reboot.
+        The loading task will be created by API.
+
+        Args:
+            image_path: A string, path to firmware image
+            boot_action: A string, boot action following the upgrade
+
+        Returns:
+            Output: A string, status and info
+                status: True or False, firmware auto-update status
```

Comment:
@sujinmkang suggest to use status codes:
```
status_installed=1
status_updated=2
status_scheduled=3
```

Reply:
@nazariig the original idea was for status to indicate the auto-update performed status which means performed successfully or not. For info, it gives more detail how the update has been performed. And platform vendor can return any information for the failure that is why the info was open.
But we can define as below. do you have any other failure cause to be added?
------------------------------------------
status           |   info
------------------------------------------
True              |  Installed = 1
True              |  Updated = 2 
True              |  Scheduled = 3
False             |  NS_BootType = -1
False             |  ImageError  = -2
False             |  ExecutionFail (or exception) = -3
--------------------------------------------